spinoso-time/src/time/tzrs/math.rs

```diff
@@ -81,225 +86,309 @@ impl Time {
+    /// Subtration — Subtracts the given u32 from _time_ and returns that value
+    /// as a new Time object.
+    ///
+    /// # Errors
+    ///
+    /// If this function attempts to overflow the the number of seconds as an
+    /// i64 then a [`TimeError`] will be returned.
+    pub fn checked_sub_u32(self, seconds: u32) -> Result<Self, TimeError> {
+        self.checked_sub_u64(u64::from(seconds))
     }
-}
 
-impl Sub<i64> for Time {
-    type Output = Self;
-
-    fn sub(self, seconds: i64) -> Self::Output {
-        self + -seconds
+    /// Subtration — Subtracts the given i64 from _time_ and returns that value
+    /// as a new Time object.
+    ///
+    /// # Errors
+    ///
+    /// If this function attempts to overflow the the number of seconds as an
+    /// i64 then a [`TimeError`] will be returned.
+    pub fn checked_sub_i64(self, seconds: i64) -> Result<Self, TimeError> {
+        if seconds.is_negative() {
+            let seconds = seconds
+                .checked_neg()
+                .and_then(|secs| u64::try_from(secs).ok())
+                .ok_or(IntOverflowError)?;
+            self.checked_add_u64(seconds)
+        } else {
+            let seconds = u64::try_from(seconds).map_err(|_| IntOverflowError)?;
+            self.checked_sub_u64(seconds)
+        }
     }
-}
-
-impl Sub<u64> for Time {
-    type Output = Self;
 
-    fn sub(self, seconds: u64) -> Self::Output {
+    /// Subtration — Subtracts the given u64 from _time_ and returns that value
+    /// as a new Time object.
+    ///
+    /// # Errors
+    ///
+    /// If this function attempts to overflow the the number of seconds as an
+    /// i64 then a [`TimeError`] will be returned.
+    pub fn checked_sub_u64(self, seconds: u64) -> Result<Self, TimeError> {
         let duration = Duration::from_secs(seconds);
-        self - duration
+        self.checked_sub(duration)
     }
-}
-
-impl Sub<f32> for Time {
-    type Output = Self;
 
-    fn sub(self, seconds: f32) -> Self::Output {
-        self - f64::from(seconds)
+    /// Subtration — Subtracts the given f32 as fraction seconds from _time_
+    /// and returns that value as a new Time object.
+    ///
+    /// # Errors
+    ///
+    /// If this function attempts to overflow the the number of seconds as an
+    /// i64 then a [`TimeError`] will be returned.
+    pub fn checked_sub_f32(self, seconds: f32) -> Result<Self, TimeError> {
+        self.checked_sub_f64(f64::from(seconds))
     }
-}
 
-impl Sub<f64> for Time {
-    type Output = Self;
-
-    fn sub(self, seconds: f64) -> Self::Output {
-        self + -seconds
+    /// Subtration — Subtracts the given f64 as fraction seconds from _time_
+    /// and returns that value as a new Time object.
+    ///
+    /// # Errors
+    ///
+    /// If this function attempts to overflow the the number of seconds as an
+    /// i64 then a [`TimeError`] will be returned.
+    pub fn checked_sub_f64(self, seconds: f64) -> Result<Self, TimeError> {
```

Comment:
like `checked_add_f64` above, I think we need to check for `NaN` and inifinities and return `TimeError` to avoid panics.